server.js

```diff
@@ -1,23 +1,74 @@
 const express = require("express");
 const cors = require("cors");
+const { response, request } = require("express");
 
 const app = express();
-
+const bodyParser = require("body-parser")
 app.use(cors());
+app.use(express.json())
+app.use(bodyParser.json())
+//data
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
+//---data
+
+
+
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
+app.get("/messages/search", (request, response) => {
+  const result = request.query.text;
+  let searchResult = messages;
+  if (result) {
+    searchResult.filter(item => item.text.includes(result))
+  }
+
+  response.send(searchResult)
```

Comment:
filter works well here, but at the moment you're sending `searchResult` back no matter what, which is always going to be the unfiltered array as you've set it to messages. you'll need to assign your result on line 31 to a variable and return that in the response